### PR TITLE
[synthetics] Document the `request` parameter

### LINKED ISSUE #2049: Document Request Object
## Description

https://github.com/elastic/synthetics/pull/555 has introduced a `request` object that can be used to make API requests (as available via the [Playwright API](https://playwright.dev/docs/test-api-testing#sending-api-requests-from-ui-tests)) independently of the browser interactions (for example to get authentication credentials / tokens).

## Collaboration

- The docs team will lead producing the content

**Contact Person:** @andrewvc 

## Suggested Target Release

This has been made available in Synthetics Agent [beta.31](https://github.com/elastic/synthetics/releases/tag/v1.0.0-beta.31), so although not strictly part of the stack release, should ideally added to coincide with 8.4.0.

## Stakeholders

@paulb-elastic @andrewvc

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Closes #2049 

## Overview

Adds documentation on the new `request` parameter that allows users to make API requests independently of the browser interactions. For example, to get authentication credentials or tokens in service of a browser-based test.

## For the reviewer

@paulb-elastic @andrewvc:
- Does it make sense to mention this initially [when introducing `journey`](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html#synthetics-create-journey)? 
- Does it make sense to add the examples to [Working with params and secrets](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-params-secrets.html#synthetics-request-param)? From what I've read I _think_ requesting credentials/tokens is an important use case, but I'm not sure if it makes sense to also document this functionality outside the context of params/secrets.

## To do

- [x] @colleenmcginnis @paulb-elastic @andrewvc agree on how info on the `request` parameter fits into the current docs
- [x] @colleenmcginnis refine language
- [x] @paulb-elastic @andrewvc review complete draft
- [x]  @colleenmcginnis clean up for publication
    - [x]  address feedback
    - [x] lint content

### CODE REVIEW

File: docs/en/observability/synthetics-params-secrets.asciidoc
Comment (paulb-elastic) [inline docs/en/observability/synthetics-params-secrets.asciidoc:96]: Not sure I fully understood the meaning of this.

Was thinking something like
> writing plain HTTP requests that are separate to the browser-based test.

**Comment by paulb-elastic** [comment]
@colleenmcginnis 

> Does it make sense to add the examples to [Working with params and secrets](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-params-secrets.html#synthetics-request-param)?

I agree, it seems a little odd for it to go in the [params and secrets](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-params-secrets.html#synthetics-request-param) section as it's out of context for the rest of that page.

I think it might be better on the [write a test](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html) page.  I think the reference to `request` you've already got there makes sense too, but maybe move the block from [here](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-params-secrets.html#synthetics-request-param) to that page (maybe between [add steps](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-params-secrets.html#synthetics-request-param) and [setup global state](https://observability-docs_2071.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html#before-after))?.

File: docs/en/observability/synthetics-params-secrets.asciidoc
Comment (colleenmcginnis) [reply docs/en/observability/synthetics-params-secrets.asciidoc:96]: This was derived from https://github.com/elastic/synthetics/pull/555.

Comment (paulb-elastic) [reply docs/en/observability/synthetics-params-secrets.asciidoc]: @andrewvc was this what you intended (it wasn't clear to me)?

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-create-test.asciidoc:187]: Should we link to the Playwright docs here?

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-create-test.asciidoc:190]: Waiting on confirmation from @andrewvc on this wording.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-create-test.asciidoc:193]: Do these examples look ok?

**Review by colleenmcginnis — COMMENTED** [review]
This is ready for review -- a few questions/comments below.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (andrewvc) [reply docs/en/observability/synthetics-create-test.asciidoc:190]: perfect

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (andrewvc) [inline docs/en/observability/synthetics-create-test.asciidoc:244]: This script also breaks, with the error below. I think at some point you changed some hard coded values to vars but didn't update that everywhere.

```
Journey: Retrieve and use HTTP Headers
   ✖  Step: 'Retrieve API key' failed (239 ms)
   ---
      stack: |-
        Error: expect(received).toEqual(expected) // deep equality

        Expected: "some api key"
        Received: "Incorrect Base64 data try: SFRUUEJJTiBpcyBhd2Vzb21l"
            at Step.callback (/home/andrewvc/projects/synthetics-demo/todos/synthetics-tests/test.ts:12:24)
```

**Review by andrewvc — DISMISSED** [review]
I like the new docs but found some bugs!

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [reply docs/en/observability/synthetics-create-test.asciidoc:193]: Thanks @andrewvc. These examples were provided by @awahab07 in https://github.com/elastic/observability-docs/issues/2049#issuecomment-1201343573.  Should I remove them from here for the launch and open an issue to track incorporating what they illustrate into https://github.com/elastic/synthetics-demo instead?

**Review by paulb-elastic — APPROVED** [review]
Agree that the example maybe does look a bit complex as there are other things in there about decoding that aren’t necessarily related to the `request`.

If the purpose of the docs is to show how a request can be made and they used in the script (irrespective of having an example in our demos are at some point), maybe a simpler example would be useful.

I’ll defer to @andrewvc’s suggestions here, but here is an example I put together for something recently (it makes a request for a quote of the day, then uses this in a Google search):
```
step("Google Page", async () => {
  // HTTP request for the quote
  const resp = await  request.get("http://quotes.stormconsultancy.co.uk/random.json");
  // parse the response
  quote = JSON.parse((await resp.body()).toString()).quote;
  // visit Google with the quote
  await page.goto('https://www.google.com?q=' + quote);
});
```
(Maybe this is too simple and not a realistic enough scenario, so feel free to use it or not)

I'll LGTM and defer to Andrew's suggestion regarding the specifics of the example.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:95]: We want to mention how this is different from the other request objects that is exposed by Playwright. 

```
1. request
2. context.request
3. page.request
```

We should refer to this docs - https://playwright.dev/docs/api/class-apirequestcontext

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-create-test.asciidoc:187]: This part of the docs is confusing it tells how user would use `playwright.request` to perform HTTP requests.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (andrewvc) [reply docs/en/observability/synthetics-create-test.asciidoc:193]: Yes, I'm OK with that

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-create-test.asciidoc:201]: >We want to mention how this is different from the other request objects that is exposed by Playwright.

@vigneshshanmugam is this what you had in mind? Are there other differences?

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-create-test.asciidoc:198]: >We want to mention how this is different from the other request objects that is exposed by Playwright.

@vigneshshanmugam is this what you had in mind? Are there other differences?

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [reply docs/en/observability/synthetics-create-test.asciidoc]: Tracking in https://github.com/elastic/observability-docs/issues/2115.

File: docs/en/observability/synthetics-params-secrets.asciidoc
Comment (colleenmcginnis) [reply docs/en/observability/synthetics-params-secrets.asciidoc]: Confirmed https://github.com/elastic/observability-docs/pull/2071#discussion_r949526167.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (andrewvc) [reply docs/en/observability/synthetics-create-test.asciidoc:95]: ++, I think a link out is plenty, as long as it's clear which part of the docs applies.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-create-test.asciidoc:198]: I was thinking something along the lines of

1. Top level `request` object exposed by synthetics have its own isolated cookie storage and its different from the  `context.request` and `page.request` as they shares cookie storage with the corresponding [BrowserContext](https://playwright.dev/docs/api/class-browsercontext).
2. If you want to control the creation of this request object - https://playwright.dev/docs/api/class-apirequest#api-request-new-context, you can do so by passing options via `--playwright-options` or `synthetics.config.ts` file.

Comment (colleenmcginnis) [reply docs/en/observability/synthetics-create-test.asciidoc]: Thanks @vigneshshanmugam. How does this look? https://github.com/elastic/observability-docs/pull/2071/commits/0fc3fee31f57ace3777ca55b3c20ef339831e164

Here's what the whole section looks like:

Screen Shot 2022-08-23 at 2 28 25 PM

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-create-test.asciidoc]: Looks great, Thanks @colleenmcginnis

**Pull request merged**
